### ISSUE TITLE
Feature request: function cupy.unique

### ISSUE THREAD

**Issue by aje** [cupy/cupy#813]
I need the function `cupy.unique` which currently does not exist in release v4.0.0b1. There exists a function `numpy.unique` https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.unique.html

For information, the reason I need it in cupy is that I'm working on a code that can work both with cupy and numpy. If cupy is available, "np" is overwritten to be cupy. But my code doesn't work since cupy do not contain the `unique` function.

Or as a workaround, do you know another way working both in numpy and cupy to obtain the list of unique elements of an array efficiently?

**Comment by kmaehashi** [comment]
Hi, thank you for using CuPy!
So far, you can implement unique function by using another functions. For example:

```py
def unique(xp, arr):
    arr = arr.flatten()
    arr.sort()
    flags = xp.concatenate((xp.array([True]), arr[1:] != arr[:-1]))
    return arr[flags]
```

This is the way how numpy finds unique elements.  https://github.com/numpy/numpy/blob/v1.13.0/numpy/lib/arraysetops.py#L276-L282

**Comment by aje** [comment]
Thanks, it works fine

**Comment by kmaehashi** [comment]
Implemented in #1140.
